torch/utils/data/dataloader.py

```diff
@@ -1470,14 +1475,14 @@ def _next_data(self):
             if idx != self._rcvd_idx:
                 if not self._in_order:
                     # don't store it for later, process now
-                    del self._task_info[idx]
-                    return self._process_data(data)
+                    worker_id = self._task_info.pop(idx)[0]
```

Comment:
Thinking out loud: in the case where a worker blocks indefinitely (or an extremely long time), I worry that _task_info will grow unbounded, but I believe this won't happen because it's being popped immediately when it's received

Reply:
Added a bit more to the comment to highlight that the entry is delete immediately